bazel/ray_deps_setup.bzl

```diff
@@ -159,15 +159,6 @@ def ray_deps_setup():
         sha256 = "97e70364e9249702246c0e9444bccdc4b847bed1eb03c5a3ece4f83dfe6abc44",
     )
 
-    auto_http_archive(
-        # This rule is used by @com_github_nelhage_rules_boost and
-        # declaring it here allows us to avoid patching the latter.
-        name = "boost",
-        build_file = "@com_github_nelhage_rules_boost//:BUILD.boost",
-        sha256 = "71feeed900fbccca04a3b4f2f84a7c217186f28a940ed8b7ed4725986baf99fa",
-        url = "https://boostorg.jfrog.io/artifactory/main/release/1.81.0/source/boost_1_81_0.tar.bz2",
```

Comment:
I can you can also do the same by pointing it to:

https://mirror.bazel.build/boostorg.jfrog.io/artifactory/main/release/1.81.0/source/boost_1_81_0.tar.gz

so that the version pin is explicitly?

it is weird that the sha256 are different from

https://github.com/nelhage/rules_boost/blob/57c99395e15720e287471d79178d36a85b64d6f6/boost/boost.bzl#L210

seems that it should be the same url.

Reply:
ah, this one was use bz2, and that one is using tgz.

Reply:
I think unless we want to pin a different version, we should piggy back on the version `com_github_nelhage_rules_boost` pick and remove the management of this dependency on our side. We end up to be the only person on the internet having this issue by doing something custom.